### PR TITLE
Migrate remaining countries to ObservedHolidayBase

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Migrate remaining countries to `ObservedHolidayBase` (Barbados, Bulgaria, Burundi, Hong Kong, Vanuatu).

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/bulgaria.py [hunk 1]
```diff
@@ -110,19 +123,14 @@ def _populate_public_holidays(self):
...
-                    continue
-                dt_observed = dt + td(days=+2 if self._is_saturday(dt) else +1)
-                while dt_observed in self:
-                    dt_observed += td(days=+1)
-                for name in self.get_list(dt):
-                    if name not in excluded_names:
-                        self._add_holiday(self.tr("%s (почивен ден)") % name, dt_observed)
+        if self.observed:
+            self.excluded_names = {self.tr("Велика събота"), self.tr("Великден")}
+            self._populate_observed(dts_observed)
```
Comment (arkid15r) [inline holidays/countries/bulgaria.py:128]: It doesn't need to be an instance variable. Just pass it to the `_populate_observed` (preferably inlined):

```suggestion
            self._populate_observed(
                dts_observed, excluded_names={self.tr("Велика събота"), self.tr("Великден")}
            )
```

If the holiday name based exclusion is going to be used somewhere else too I'd suggest making `excluded_names` logic a part of `ObservedHolidayBase`.

File: holidays/countries/bulgaria.py [hunk 2]
```diff
@@ -43,13 +44,25 @@ class Bulgaria(HolidayBase, ChristianHolidays, InternationalHolidays):
...
 
     def __init__(self, *args, **kwargs):
         ChristianHolidays.__init__(self, JULIAN_REVISED_CALENDAR)
         InternationalHolidays.__init__(self)
-        super().__init__(*args, **kwargs)
+        super().__init__(
+            observed_rule=SAT_SUN_TO_NEXT_WORKDAY, observed_since=2017, *args, **kwargs
+        )
+
+    def _populate_observed(self, dts: Set[date], multiple: bool = False) -> None:
```
Comment (arkid15r) [inline holidays/countries/bulgaria.py:59]: `multiple` is unused

File: holidays/countries/barbados.py [hunk 3]
```diff
@@ -9,15 +9,17 @@
...
-from datetime import timedelta as td
-
 from holidays.calendars.gregorian import JAN, JUL
 from holidays.groups import ChristianHolidays, InternationalHolidays
-from holidays.holiday_base import HolidayBase
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    SUN_TO_NEXT_MON,
+    SUN_TO_NEXT_TUE,
+    MON_TO_NEXT_TUE,
```
Comment (arkid15r) [inline holidays/countries/barbados.py:18]:
```suggestion
    ObservedHolidayBase,
    MON_TO_NEXT_TUE,
    SUN_TO_NEXT_MON,
    SUN_TO_NEXT_TUE,
```

**Review by arkid15r — COMMENTED** [review]
Some minor comments to address:

File: holidays/countries/bulgaria.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/bulgaria.py]: Done.

File: holidays/countries/bulgaria.py (hunk 2 above)
Comment (KJhellico) [reply holidays/countries/bulgaria.py]: I just left parent class method signature unchanged 👀

**Pull request merged**